homeassistant/components/tplink/entity.py

```diff
@@ -57,24 +78,201 @@ async def _async_wrap(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> None:
     return _async_wrap
 
 
-class CoordinatedTPLinkEntity(CoordinatorEntity[TPLinkDataUpdateCoordinator]):
+class CoordinatedTPLinkEntity(CoordinatorEntity[TPLinkDataUpdateCoordinator], ABC):
     """Common base class for all coordinated tplink entities."""
 
     _attr_has_entity_name = True
 
     def __init__(
-        self, device: Device, coordinator: TPLinkDataUpdateCoordinator
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        feature: Feature | None = None,
+        parent: Device | None = None,
     ) -> None:
-        """Initialize the switch."""
+        """Initialize the entity."""
         super().__init__(coordinator)
         self.device: Device = device
-        self._attr_unique_id = device.device_id
+        self._feature = feature
         self._attr_device_info = DeviceInfo(
-            connections={(dr.CONNECTION_NETWORK_MAC, device.mac)},
+            # connections={(dr.CONNECTION_NETWORK_MAC, device.mac)},
```

Comment:
Open question if this is needed when we have identifiers? If set for child devices, all of them will be presented by a single device.

Reply:
If its a child device we shouldn't set it and instead use `via_device`

Reply:
Done. `via_device` is now used when we have a parent device, otherwise the connection is created.